app/models/ability.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize user
```

Comment:
đối tượng user này lấy từ đâu ?

Reply:
- CanCan expects a current_user method to exist in the controller.
=> đối tượng user này lấy từ current_user (từ devise nếu có sử dụng hoặc method current_user tự định nghĩa)